core/trino-main/src/main/java/io/trino/sql/planner/RelationPlanner.java

```diff
@@ -291,6 +291,29 @@ public RelationPlan addRowFilters(Table node, RelationPlan plan, Function<Expres
         return new RelationPlan(planBuilder.getRoot(), plan.getScope(), plan.getFieldMappings(), outerContext);
     }
 
+    public RelationPlan addCheckConstraints(List<Expression> filters, Table node, RelationPlan plan, Function<Expression, Expression> predicateTransformation, Function<Table, Scope> accessControlScope)
```

Comment:
What's "predicate transformation"?

Reply:
That's a function inserting an expression to fail when the constraint is violated. Do you have any suggestions for the new variable name? 
```java
    private static Function<Expression, Expression> failIfCheckConstraintIsNotMet(Metadata metadata, Session session)
    {
        return predicate -> new IfExpression(
                // When predicate evaluates to UNKNOWN (e.g. NULL > 100), it should not violate the check constraint.
                new CoalesceExpression(predicate, TRUE_LITERAL),
                TRUE_LITERAL,
                new Cast(failFunction(metadata, session, CONSTRAINT_VIOLATION, "Check constraint violation: " + predicate), toSqlType(BOOLEAN)));
    }
```

Reply:
I'm not sure that abstraction is even necessary. Why not inline that logic directly in addCheckConstraints?

Reply:
I.e., just modify addCheckConstraints to do this:


```java
public RelationPlan addCheckConstraints(List<Expression> constraints, Table node, RelationPlan plan, Function<Table, Scope> accessControlScope)
{
    ...
    for (Expression constraint : constraints) {
        ...

        Expression predicate = new IfExpression(
                // When predicate evaluates to UNKNOWN (e.g. NULL > 100), it should not violate the check constraint.
                new CoalesceExpression(coerceIfNecessary(analysis, constraint, planBuilder.rewrite(constraint)), TRUE_LITERAL),
                TRUE_LITERAL,
                new Cast(failFunction(plannerContext.getMetadata(), session, CONSTRAINT_VIOLATION, "Check constraint violation: " + predicate), toSqlType(BOOLEAN)));

        planBuilder = planBuilder.withNewRoot(new FilterNode(
                idAllocator.getNextId(),
                planBuilder.getRoot(),
                predicate));
    }

    ...
}
```

Reply:
@martint Agree with inline. Updated.